tools/toolchains/arm.py

```diff
@@ -175,32 +177,53 @@ def compile_c(self, source, object, includes):
     def compile_cpp(self, source, object, includes):
         return self.compile(self.cppc, source, object, includes)
 
-    @hook_tool
-    def link(self, output, objects, libraries, lib_dirs, mem_map):
-        map_file = splitext(output)[0] + ".map"
-        if len(lib_dirs):
-            args = ["-o", output, "--userlibpath", ",".join(lib_dirs), "--info=totals", "--map", "--list=%s" % map_file]
-        else:
-            args = ["-o", output, "--info=totals", "--map", "--list=%s" % map_file]
-
-        args.extend(self.flags['ld'])
-
-        if mem_map:
-            args.extend(["--scatter", mem_map])
+    def correct_scatter_shebang(self, scatter_file):
+        """Correct the shebang at the top of a scatter file.
+
+        Positional arguments:
+        scatter_file -- the scatter file to correct
+
+        Return:
+        The location of the correct scatter file
+
+        Side Effects:
+        This method MAY write a new scatter file to disk
+        """
+        with open(scatter_file, "rb") as input:
+            lines = input.readlines()
+            if  (lines[0].startswith(self.SHEBANG) or
```

Comment:
should the line be normalized before using this check? That way extra spaces won't cause this comparison to fail

Reply:
The comparison failing would not be that bad: it would just emit another linker script with the correct shebang. I'm not sure I want to make that check complicated, but you're right that it can be improved to reduce the frequency of emitting linker scripts.

Reply:
Good point. I didn't realize there is no harm done if the comparison fails. I agree, it would be good to keep it simple like it is now.